src/expr.rs

```diff
@@ -1194,6 +1194,63 @@ pub(crate) fn is_simple_block(
         && attrs.map_or(true, |a| a.is_empty())
 }
 
+pub(crate) fn contains_curly_block(expr: &ast::Expr) -> bool {
```

Comment:
I don't know if `contains_curly_block` is the right name for this function. Are you trying to add space around anything with curly braces or just all statements?

Reply:
I'm open to suggestions for names.

Here's the "problem" that I've run into: I've been reviewing code that will have 50 lines without a single line between them. If one looks at any popular open-source repo (including rustfmt), one can see many more empty lines, breaking code up into logical chunks like paragraphs in prose. So the question is, how can a formatter that doesn't have logic context of the code add some of this spacing? Naturally, new scopes (demarcated by curly braces) provide an opportunity to add spacing.

So maybe the name should be something like `has_new_scope`? Or something along those lines...